src/main/java/dp/xlsx/Group.java

```diff
@@ -71,7 +81,30 @@ Observation getObservation(String time) {
 
     @Override
     public int hashCode() {
-        return getGroupValues().hashCode();
+        /**
+         * Generate a string representation of the group by concatinating the string values of each
+         * {@link DimensionData}. The hashcode of this string is the hashcode for this object.
+         *
+         * Explanation:
+         * The previous hashcode implementation for {@link Group} used the hash code of the
+         * {@link Group}{@link #getGroupValues()} but this hashcode was not specific enough when two
+         * {@link DimensionData} objects had very similar values.
+         *
+         * The Group objects created from the following CSV rows although different were generating the same hashcode
+         * value:
+         *
+         *      ,.,1978-to-2018-19,1978 to 2018-19,K02000001,United Kingdom,18,18,gross-income,Gross income,1990s,1990s
+         *      ,.,1978-to-2018-19,1978 to 2018-19,K02000001,United Kingdom,19,19,gross-income,Gross income,1980s,1980s
+         *
+         * This hashcode collision resulted in the {@link V4File} incorrecting parsing the V4 file leading to missing
+         * rows or rows with incorrect observations being assigned.
+         * This slightly wierd approach attempts to address that problem. See GroupTest for me details.
+         */
+        return StringUtils.join(getGroupValues()
+                .stream()
+                .map(val -> val.toString())
+                .collect(Collectors.toList()), "|")
+                .hashCode();
```

Comment:
I may have this totally wrong (due to not being a java programmer), but on this link:
    https://docs.oracle.com/javase/10/docs/api/java/lang/Object.html#hashCode()
The 3rd bullet point under:
    The general contract of hashCode is:
Seems to me to imply that you may still end up with a non unique hash code.

Reply:
Yes thats true. I'm not sure to what extend this will fix the current problem or if its just delaying further collisions.  As part of testing this I actually ran the entire dataset cvs and it is now correctly parsing the data which gives me reasonable confidence as its ~4k rows which is fairly small but if there further issues I'd hope we would have seen evidence of this.

The only alternative I can see it to completely rewrite this section to replace this HashMap with a List and write a find/filter method to achieve the look up.

Reply:
I think the changes will *probably* make the groups unique across different datasets, though by relying on the hashcode for comparison we cannot be 100% sure. 

I think we can still utilise a map (and hashcodes) for quick lookups, but can use a list of groups as the key of the map. If there is a clash of hashcodes when the map is built, both groups get added to the same key/list. When we lookup a group, we would expect only one entry in the list. If there are two, we can then use a comparator to determine the exact group.

Reply:
Update:

turns out the  `equals` method in `Group` hasn't been implemented correctly - its comparing hashcodes. I think if I update this properly it should resolve this problem properly.